newsfragments/4196.changed.md

```diff
@@ -0,0 +1,2 @@
+Update PyType::name and add PyType::{module,fully_qualified_name}
+to match https://peps.python.org/pep-0737.
```

Comment:
I think worth splitting the additions out from this file into a second fragment `4196.added.md`. Then we can make more detail on the `PyType::name` change:

```suggestion
Change `PyType::name` to always match Python `__name__`.
```